pkgs/development/libraries/tpm2-tss/default.nix

```diff
@@ -3,6 +3,14 @@
 , openssl, json_c, curl, libgcrypt
 , cmocka, uthash, ibm-sw-tpm2, iproute2, procps, which
 }:
+let
+  # Since tpm2-tss is a build input to our (full) systemd build
+  # we can't have packages the closure of tpm2-tss that depend on systemd itself.
+  # Since procps is only used during tests (which are executed within the Nix sandbox)
+  # we can safely pass a version that doesn't support systemd specific information.
+  procpsWithoutSystemd = procps.override { withSystemd = false; };
+  procps_pkg = if stdenv.isLinux then procpsWithoutSystemd else procps;
```

Comment:
I don't think it would hurt to use the withSystemd = false; version everywhere, since that flag is probably only relevant on linux anyway. Let's simplify this?

Reply:
That wouldn't work was it is a different package on not-linux and thus we can't disable it in that case. I shouldn't have removed the comment that I had here before :)

Reply:
Ah, this makes sense indeed. I didn't consider that procps would be a different derivation on !linux, but in hindsight it's pretty obvious. Suggested reworded comment block:

```
# Avoid a circular dependency on Linux systems (systemd depends on tpm2-tss,
# tpm2-tss tests depend on procps, procps depends on systemd by default). This
# needs to be conditional based on isLinux because procps for other systems
# might not support the withSystemd option.
```